DnDCharacterSheet/Repositories/CoinRepository.cs

```diff
@@ -0,0 +1,58 @@
+﻿using DnDCharacterSheet;
+using Interfaces;
+using Models;
+
+namespace Repositories
+{
+    public class CoinRepository(AppDbContext dbContext) : ICoinRepository
+    {
+        private readonly AppDbContext _dbContext = dbContext;
+
+        public Coin AddCoin(Coin coin)
+        {
+            _dbContext.Coins.Add(coin);
+            _dbContext.SaveChanges();
+            return coin;
+        }
+
+        public bool DeleteCoin(long id)
+        {
+            var coinToDelete = GetCoinById(id);
+            if (coinToDelete == null)
```

Comment:
Important : Take a look into pattern matching, but my suggestion is to use (T is null / is not null) it has clear advantages and should be preferred when possible, == is an overloadable operator